glottolog3/views.py

```diff
@@ -318,6 +321,47 @@ def bpsearch(request):
     return {'message': message, 'params': params, 'languoids': languoids,
         'map': map_, 'countries': countries}
 
+@view_config(
+        route_name='glottolog.add_identifier',
+        request_method='POST', 
+        renderer='json')
+def add_identifier(request):
+    # TODO: Add validity checks for parameters and unit tests
+
+    gcode = request.json_body['glottocode']
+    lang = request.json_body['language']
+    name = request.json_body['name']
+    type = request.json_body['type']
+    desc = request.json_body['description']
+
+    languoid = DBSession.query(Language) \
+                        .filter_by(id='{0}'.format(gcode)) \
+                        .first()
+
+    identifier = Identifier(
+        (name, type, desc, lang),
+        id='{0}-{1}-{2}-{3}'.format(
+        slug(name), slug(type), slug(desc or ''), lang),
+        name=name,
+        type=type,
+        description=desc,
+        lang=lang)
+
+    try:
+        DBSession.add(identifier)
+        DBSession.add(
+            LanguageIdentifier(language=languoid, identifier=identifier))
+        DBSession.flush()
+    except exc.SQLAlchemyError as e:
+        DBSession.rollback()
+        return {'message': 'ERROR: {}'.format(e)}
```

Comment:
Ioana uses some good conventions in #12, let's change this to just be `{ 'error': '{}'.format(e) }`